### ISSUE TITLE
C++ syntax highlighting is broken

### ISSUE THREAD

**Issue by dimmo** [Microsoft/vscode#18242]
- VSCode Version: Code 1.8.1 (ee428b0eead68bf0fb99ab5fdc4439be227b6281, 2016-12-19T14:41:20.664Z)
- OS Version: Linux x64 4.4.0-57-generic
- Extensions:

|Extension|Author|Version|
|---|---|---|
|githistory|donjayamanne|0.1.2|
|python|donjayamanne|0.5.5|
|cpptools|ms-vscode|0.9.3|
|csharp|ms-vscode|1.6.2|

---

Steps to Reproduce:

Create a.cpp file:
```
class Foo {
    long Sum() {
        return 1;
    }
    long A() {
        return true ?
            Sum() : 0;
    }
};

class Bar {
};

int main() {
    Bar bar;
    return 0;
}
```

Expected behavior: both words "class" are highlighted (blue).
Actual behavior: first word "class" is blue, second is gray.

![vs_code_syntax_2017-01-07_01-55-54](https://cloud.githubusercontent.com/assets/3662090/21732744/ca99225e-d47c-11e6-8259-0faa39d98410.png)

**Comment by Swiiip** [comment]
On VSCode 1.9.0-Insider with no C++ extension installed, it works correctly

[http://img11.hostingpics.net/pics/725424Capturedecran20170107a135044.png](url)

**Comment by sandy081** [comment]
Moved to Cpp tools repository

**Comment by sean-mcmanus** [comment]
@dimmo This was fixed in VS Code 1.9 when they started using the newer atom/language-c (it wasn't a cpptools bug).
